scripts/api.py

```diff
@@ -205,9 +206,59 @@ async def detect(
         controlnet_module = global_state.reverse_preprocessor_aliases.get(controlnet_module, controlnet_module)
 
         if controlnet_module not in global_state.cn_preprocessor_modules:
-            return {"images": [], "info": "Module not available"}
+            raise HTTPException(
+                status_code=500, detail="Module not available")
+        
+        available_modules = [
```

Comment:
is there a reason we need to duplicate the preprocessors keys here? The exception above should handle all cases.

Reply:
Ah it seems to be an artifact from older code. In any case, please consider removing it.

Reply:
You're right, it was pulled in with the rebase for some reason!
I'll remove it.